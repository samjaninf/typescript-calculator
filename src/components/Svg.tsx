import { StyleAttribute, css, media } from 'glamor'
import { h } from 'preact'
import { small } from '../styles/breakpoints'

type SvgProps = {
  children?: JSX.Element,
}

const viewBox: string = '0 0 64 64'
const svg: StyleAttribute = css(
  {
    height: 16,
    marginRight: '1em',
    marginTop: 0,
    width: 16,
  },
  media(small, { marginTop: -6 }),
)

const Svg = ({ children }: SvgProps): JSX.Element => (
  <div className={svg.toString()}>
    <svg
      enableBackground={`new ${viewBox}`}
      viewBox={viewBox}
      xmlns="http://www.w3.org/2000/svg"
    >
      {children}
    </svg>
  </div>
)

export default Svg
