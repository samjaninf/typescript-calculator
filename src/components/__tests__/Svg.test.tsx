import { h } from 'preact'
import Svg from '../Svg'
import snapshot from './utils/snapshot'

describe('Svg', () => {
  it('renders', () => snapshot(Svg))
  it('renders children', () => snapshot(Svg, { children: <g /> }))
})
