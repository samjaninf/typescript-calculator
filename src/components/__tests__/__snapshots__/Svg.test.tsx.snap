// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg renders 1`] = `
"<div class=\\"css-k5inyr\\">
  <svg enableBackground=\\"new 0 0 64 64\\" viewBox=\\"0 0 64 64\\" xmlns=\\"http://www.w3.org/2000/svg\\"></svg>
</div>"
`;

exports[`Svg renders children 1`] = `
"<div class=\\"css-k5inyr\\">
  <svg enableBackground=\\"new 0 0 64 64\\" viewBox=\\"0 0 64 64\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <g></g>
  </svg>
</div>"
`;
